### PR TITLE
Datetime formatting

### PR DESCRIPTION
**Pull request by finnss**
skjermbilde 2017-10-24 kl 21 30 18

### CODE REVIEW

**Comment by ekmartin** [comment]
`17:15 til 20:15 - tirsdag 24. oktober` kanskje?
eller motsatt

File: app/routes/events/components/EventDetail/index.js [hunk 1]
```diff
@@ -129,6 +130,20 @@ export default class EventDetail extends Component<Props> {
       ? () => unfollow(event.isUserFollowing.id, event.id)
       : () => follow(currentUser.id, event.id);
 
+    const formatTime = (time: moment) => {
+      const dateTime = moment(time);
+
+      let format;
+
+      if (moment().year() === Number(dateTime.format('YYYY'))) {
```
Comment (ekmartin) [inline app/routes/events/components/EventDetail/index.js]: moment().isSame(dateTime, 'year')

**Comment by finnss** [comment]
`tirsdag 24. oktober 17:15 til 20:15`?

**Comment by finnss** [comment]
Blir mye tekst
skjermbilde 2017-10-24 kl 20 19 30

File: app/routes/events/components/EventDetail/index.js [hunk 2]
```diff
@@ -129,6 +130,20 @@ export default class EventDetail extends Component<Props> {
       ? () => unfollow(event.isUserFollowing.id, event.id)
       : () => follow(currentUser.id, event.id);
 
+    const formatTime = (time: moment) => {
+      const dateTime = moment(time);
```
Comment (hanswilw) [inline app/routes/events/components/EventDetail/index.js]: You dont need to moment(time), time is already a moment object :) startTime, endTime are moment already as well

Comment (finnss) [reply app/routes/events/components/EventDetail/index.js]: That's what I thought too but that wasn't the case for `event.paymentDueDate` and some other the others

Comment (hanswilw) [reply app/routes/events/components/EventDetail/index.js]: Then fix the transform for those dates in the same way as startTime instead :)

Comment (finnss) [reply app/routes/events/components/EventDetail/index.js]: So I guess it's incorrect flow to say `(time: moment)` but my code crashed when assuming the parameter was already a `moment`.

File: app/routes/events/components/EventDetail/index.js [hunk 3]
```diff
@@ -169,15 +184,10 @@ export default class EventDetail extends Component<Props> {
...
+                  <span className={styles.metaDescriptor}>Når</span>
                   <strong>
-                    <Time time={event.startTime} format="DD.MM.YYYY HH:mm" />
-                  </strong>
-                </li>
-                <li>
-                  <span className={styles.metaDescriptor}>Slutter</span>
-                  <strong>
-                    <Time time={event.endTime} format="DD.MM.YYYY HH:mm" />
+                    {formatTime(event.startTime)} til{' '}
```
Comment (odinuge) [inline app/routes/events/components/EventDetail/index.js]: What about making it a stateless component?

ala ish
```jsx
const FormatTime = ({time, format}: {time: theMomentType, format?: string})=> {
    if (!format){
        // define proper formatting
    }
   [...]

}

<FormatTime time={event.startTime} />
<FormatTime time={event.endTime} format='HH:mm' />
```

**Comment by hanswilw** [comment]
Remember to handle the case of dates over multiple days, like Åre etc.

**Comment by odinuge** [comment]
What about making it a stateless component?

ala ish
```jsx
const FormatTime = ({time, format}: {time: theMomentType, format?: string})=> {
    if (!format){
        // define proper formatting
    }
   [...]

}

<FormatTime time={event.startTime} />
<FormatTime time={event.endTime} format='HH:mm' />
```

File: app/routes/events/components/EventDetail/index.js [hunk 4]
```diff
@@ -129,6 +149,17 @@ export default class EventDetail extends Component<Props> {
       ? () => unfollow(event.isUserFollowing.id, event.id)
       : () => follow(currentUser.id, event.id);
 
+    const fromToDate = (
```
Comment (odinuge) [inline app/routes/events/components/EventDetail/index.js]: pls this one tooo

```jsx
<FromToTime from={event.startTime} to={event.endTime} />
```

**Comment by finnss** [comment]
Smart @odinuge, done!

File: app/routes/events/components/EventDetail/index.js (hunk 4 above)
Comment (finnss) [reply app/routes/events/components/EventDetail/index.js]: Really? For a one-time thing?:o

Comment (odinuge) [reply app/routes/events/components/EventDetail/index.js]: It makes reading the component a lot smoother

File: app/routes/events/components/EventDetail/index.js [hunk 5]
```diff
@@ -129,6 +149,17 @@ export default class EventDetail extends Component<Props> {
...
+      <strong>
+        <FormatTime time={event.startTime} /> -{' '}
+        {event.startTime.format('DD.MM') === event.endTime.format('DD.MM') ? (
```
Comment (ekmartin) [inline app/routes/events/components/EventDetail/index.js]: can use `isSame` here as well

File: app/routes/events/components/EventDetail/index.js (hunk 4 above)
Comment (odinuge) [reply app/routes/events/components/EventDetail/index.js]: It should also be possible to reuse it in meetings.

**Comment by finnss** [comment]
There. I think the format looks good, but it's a bit of a shame that it jumps to a new line for the last two. Is there any merit to the idea of increasing the width a little or reduce the font size or some such to avoid it?

skjermbilde 2017-10-24 kl 20 48 48

Åre:
skjermbilde 2017-10-24 kl 20 49 09

**Comment by edvardvb** [comment]
It might not be short enough for Åre, but would it help to use shortened days and months? I.e. `ons 25. okt, 19.57`

**Comment by finnss** [comment]
Appearantly the only option for shortening without hacky solutions was `on`, `to` etc. Besides, I like the full weekday string personally! Perhaps the month should be shortened though, that could be enough.

**Comment by finnss** [comment]
90% of the time only the "Når" time will be relevant for most people, so I think that should be pretty. The latest commit uses long weekdays and months for `Når` with start and end in the same day, but shortened months for the rest:


skjermbilde 2017-10-24 kl 21 15 23

Åre:
skjermbilde 2017-10-24 kl 21 17 54

**Comment by edvardvb** [comment]
Much better imo :) However, the lack of a comma after the month triggers me

**Comment by finnss** [comment]
Haha gotchu fam

skjermbilde 2017-10-24 kl 21 30 18

**Comment by odinuge** [comment]
Would be nice if you add this to `app/components/Time`, and write a few edge-case tests - so we know it works as intended. Would be sweet to have this in meetings too. :smile:
